### PR TITLE
Cope with spaces in install dir

### PR DESCRIPTION
**Pull request by james-portman**

### CODE REVIEW

File: bin/terraform [hunk 1]
```diff
@@ -3,4 +3,6 @@ set -e
 [ -n "$TFENV_DEBUG" ] && set -x
 
 program="${0##*/}"
-exec "$(dirname `which $0`)"/tfenv exec "$@"
+which=$(which "$0")
+dir=$(dirname "$which")
```
[the remark this answers is not in the document]
Comment (james-portman) [reply bin/terraform]: I did try it as a one-liner but the multiple sets of quotes were upsetting it, so I was being super safe

[the remark this answers is not in the document]
Comment (Zordrak) [reply bin/terraform]: Does my example resolve the upset ;-) ?

[the remark this answers is not in the document]
Comment (Zordrak) [reply bin/terraform]: Also, would be useful if you can devise a way to add a regression test for spaces in paths if this is explicitly to fix that issue. Then we can ensure new code does not later break spaced-paths.

[the remark this answers is not in the document]
Comment (james-portman) [reply bin/terraform]: The above suggestion does also work, updated

**Comment by james-portman** [comment]
Bump, can you merge this at least to fix the issue for now?

**Comment by Zordrak** [comment]
When not on a phone I will rebase this branch on master and merge it unless someone beats me to the rebase

**Pull request closed without merging**